src/expandableCalendar/WeekCalendar/index.tsx

```diff
@@ -1,144 +1,80 @@
 
 export interface WeekCalendarProps extends CalendarListProps {
   /** whether to have shadow/elevation for the calendar */
   allowShadow?: boolean;
   context?: any;
 }
 
-interface State {
-  items: string[];
-}
-
 /**
  * @description: Week calendar component
  * @note: Should be wrapped with 'CalendarProvider'
  * @example: https://github.com/wix/react-native-calendars/blob/master/example/src/screens/expandableCalendar.js
  */
-class WeekCalendar extends Component<WeekCalendarProps, State> {
-  static displayName = 'WeekCalendar';
-
-  static propTypes = {
-    ...CalendarList.propTypes,
-    allowShadow: PropTypes.bool
-  };
-
-  static defaultProps = {
-    firstDay: 0,
-    allowShadow: true
-  };
-
-  style = styleConstructor(this.props.theme);
-  presenter = new Presenter();
-  page = NUMBER_OF_PAGES;
-  // On Android+RTL there's an initial scroll that cause issues
-  firstAndroidRTLScrollIgnored = !applyAndroidRtlFix;
-
-  state: State = {
-    items: this.presenter.getDatesArray(this.props)
-  };
-
-  componentDidUpdate(prevProps: WeekCalendarProps) {
-    const {context, firstDay = 0} = this.props;
-    const {shouldComponentUpdate, getDatesArray, scrollToIndex} = this.presenter;
-
-    if (shouldComponentUpdate(this.props.context, prevProps.context)) {
-      if (!sameWeek(context.date, prevProps.context.date, firstDay)) {
-        // Don't update items if the new date is on the same week
-        this.setState({items: getDatesArray(this.props)});
-        scrollToIndex(false);
-      }
-    }
-  }
-
-  get containerWidth() {
-    return this.props.calendarWidth || constants.screenWidth;
-  }
-
-  getDatesArray() {
-    const array = [];
-    for (let index = -NUMBER_OF_PAGES; index <= NUMBER_OF_PAGES; index++) {
-      const d = this.getDate(index);
-      array.push(d);
-    }
-    return array;
-  }
-
-  getDate(weekIndex: number) {
-    const {current, context, firstDay = 0} = this.props;
-    const d = new XDate(current || context.date);
-    // get the first day of the week as date (for the on scroll mark)
-    let dayOfTheWeek = d.getDay();
-    if (dayOfTheWeek < firstDay && firstDay > 0) {
-      dayOfTheWeek = 7 + dayOfTheWeek;
-    }
-
-    // leave the current date in the visible week as is
-    const dd = weekIndex === 0 ? d : d.addDays(firstDay - dayOfTheWeek);
-    const newDate = dd.addWeeks(weekIndex);
-    return toMarkingFormat(newDate);
-  }
-
-  getWeekStyle = memoize((width, style) => {
-    return [{width}, style];
-  });
-
-  onDayPress = (value: DateData) => {
-    this.presenter.onDayPress(this.props.context, value);
-  };
-
-  onScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+const WeekCalendar = (props: WeekCalendarProps) => {
+  const {calendarWidth, firstDay, hideDayNames, current, markedDates, allowShadow, context, theme, style} = props;
+  const defaultStyle = useMemo(() => styleConstructor(theme), [theme]);
+  const presenter = new Presenter();
```

Comment:
I personally don't like the presenter and would love to see it as a service for pure functions (not a class), while moving all the events back to the component. What do you think?